### PR TITLE
Updated the list of SQL Errors

### PR DESCRIPTION
**Pull request by rewanth1997**
Updated the list of SQL errors in nselib/data/http-sql-errors.lst.

### CODE REVIEW

File: nselib/data/http-sql-errors.lst [hunk 1]
```diff
@@ -39,13 +46,14 @@ MySQL ODBC
 ODBC DB2
 ODBC Driver
 ODBC Error
-ODBC Microsoft Access
+ODBC Microsoft Access Driver
```
Comment (dmiller-nmap) [inline nselib/data/http-sql-errors.lst]: What is the reference for changing this? This makes the match more specific; is the word "Driver" always included?

File: nselib/data/http-sql-errors.lst [hunk 2]
```diff
@@ -56,33 +64,48 @@ PHP Warning
...
+Supplied argument is not a valid MySQL result resource in
 Supplied argument is not a valid PostgreSQL result
 Syntax error in query expression
+System.IndexOutOfRangeException
 The error occurred in
 The script whose uid is
 Type mismatch
 Unable to jump to row
-Unclosed quotation mark before the character string
+Unable to select * from
```
Comment (dmiller-nmap) [inline nselib/data/http-sql-errors.lst]: Need to %-escape the * here

File: nselib/data/http-sql-errors.lst [hunk 3]
```diff
@@ -56,33 +64,48 @@ PHP Warning
...
+Unclosed quotation mark
+Unknown column
 Unterminated string constant
 Warning: Cannot modify header information %- headers already sent
 Warning: Supplied argument is not a valid File%-Handle resource in
 Warning: mysql_query%(%)
 Warning: pg_connect%(%): Unable to connect to PostgreSQL server: FATAL
-You have an error in your SQL syntax near
+Warning: require(%)
+array_merge()
```
Comment (dmiller-nmap) [inline nselib/data/http-sql-errors.lst]: Parentheses here and throughout must be %-escaped. https://www.lua.org/manual/5.3/manual.html#6.4.1

File: nselib/data/http-sql-errors.lst [hunk 4]
```diff
@@ -39,13 +46,14 @@ MySQL ODBC
...
 ODBC Error
-ODBC Microsoft Access
+ODBC Microsoft Access Driver
 ODBC Oracle
 ODBC SQL
 ODBC SQL Server
 OLE/DB provider returned message
 ORA%-0
 ORA%-1
+ORA-01756
```
Comment (dmiller-nmap) [inline nselib/data/http-sql-errors.lst]: Need to escape the `-` here, but isn't this a subset of the earlier `ORA%-0` match?

File: nselib/data/http-sql-errors.lst (hunk 1 above)
Comment (rewanth1997) [inline nselib/data/http-sql-errors.lst]: Well, I'm unable to find the previous reference Dan. I'm will undo this change.

File: nselib/data/http-sql-errors.lst (hunk 4 above)
Comment (rewanth1997) [inline nselib/data/http-sql-errors.lst]: This is a duplicate, this will be deleted.

File: nselib/data/http-sql-errors.lst (hunk 2 above)
Comment (rewanth1997) [inline nselib/data/http-sql-errors.lst]: I thought it to be a normal text file, I will change it.

File: nselib/data/http-sql-errors.lst (hunk 3 above)
Comment (rewanth1997) [inline nselib/data/http-sql-errors.lst]: It will be corrected in my next commit.

**Comment by dmiller-nmap** [comment]
Other [Lua pattern special characters](https://www.lua.org/manual/5.3/manual.html#6.4.1) need to be escaped. I had only commented on the first of each.

**Comment by rewanth1997** [comment]
@dmiller-nmap All the special characters are escaped now. Please confirm.

**Comment by dmiller-nmap** [comment]
Looks good, please commit

**Comment by nnposter** [comment]
Please make sure that pattern metacharacters are being consistently escaped. Example:
```diff
 Supplied argument is not a valid PostgreSQL result
 Syntax error in query expression
+System.IndexOutOfRangeException
 The error occurred in
 The script whose uid is
```

**Comment by rewanth1997** [comment]
@nnposter Fix committed as https://github.com/nmap/nmap/commit/3566f2ba8b35a05dde1edbc433291b6731a69e69.
